### PR TITLE
[YNGJ-698] Use GitHub actions for CI process

### PR DESCRIPTION
**Pull request by nehaabraham**
@bklang @benlangfeld
This PR adds CI process using github actions and removes Travis CI. I tried to include a wide range of ruby and jruby versions.

### CODE REVIEW

**Review by ns-ian — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: build
+on:
+    - push
+    - pull_request
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: ['2.1.9', '2.2.10', '2.3.0', '2.4.0', '2.5.0', '2.6.0', '2.7.0', 'jruby-9.1.17.0', 'jruby-9.2.9.0', 'jruby-9.2.21.0']
```
Comment (bklang) [inline .github/workflows/build.yml:10]: Is there a specific reason to test both `jruby-9.2.9.0` and `jruby-9.2.21.0`? Historically our general practice has been to test the latest version of each minor release.

**Comment by bklang** [comment]
Other than the question about the ruby versions to test, this LGTM.

File: .github/workflows/build.yml (hunk 1 above)
Comment (nehaabraham) [reply .github/workflows/build.yml:10]: No specific reason, I can remove jruby 9.2.9.0

Comment (bklang) [reply .github/workflows/build.yml]: We are also hard coding 2.3.0, 2.4.0, 2.5.0, etc. Is there a way to say "latest 2.5.x"?

Comment (nehaabraham) [reply .github/workflows/build.yml]: Looks like if I leave out the minor release it will resolve to the latest. So if I say 2.5 it will resolve to 2.5.9

**Review by bklang — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**